### PR TITLE
adjust `account not found` error when sending a transaction

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
## Reasoning behind the pull request
- previously, when sending a transaction from a 'new' account ('new' means that it didn't have any interaction with the blockchain, that is, neither sending nor receiving a transaction) the returned error would have been 'account not found for address X and shard Y"
- it was not consistent with other situations (an address that exists in the trie but does not have any balance - would have returned other errors)
  
## Proposed changes
- return `insufficient balance` instead of `account not found`

## Testing procedure
- generate a valid transaction from an address that doesn't have any interaction with the chain. When sending it, you should receive an `insufficient balance` instead of `account not found`

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: api/groups/transactionGroup.go [hunk 1]
```diff
@@ -465,6 +468,20 @@ func (tg *transactionGroup) sendMultipleTransactions(c *gin.Context) {
...
+		return nil
+	}
+
+	// TODO: might move these errors from the process package so they can be used here as well
+
+	if !strings.Contains(err.Error(), "account not found") {
+		return err
+	}
+
+	return fmt.Errorf("insufficient balance for address %s", address)
```
Comment (iulianpascalau) [inline api/groups/transactionGroup.go:482]: this is a bad assumption that any other error other than account not found is caused by insufficient balance. I would add something like:
```
return fmt.Errorf("%w that might be caused by insufficient balance for address %s", err, address)
```

File: api/groups/transactionGroup.go [hunk 2]
```diff
@@ -465,6 +468,20 @@ func (tg *transactionGroup) sendMultipleTransactions(c *gin.Context) {
 	)
 }
 
+func adjustTxValidatorErrorIfNeeded(err error, address string) error {
```
Comment (iulianpascalau) [inline api/groups/transactionGroup.go:471]: wrapSendError?

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: api/groups/transactionGroup.go (hunk 1 above)
Comment (bogdan-rosianu) [reply api/groups/transactionGroup.go]: refactored

File: api/groups/transactionGroup.go (hunk 2 above)
Comment (bogdan-rosianu) [reply api/groups/transactionGroup.go]: refactored

File: node/node.go [hunk 3]
```diff
@@ -54,7 +54,8 @@ var log = logger.GetOrCreate("node")
 var _ facade.NodeHandler = (*Node)(nil)
 
 // Option represents a functional configuration parameter that can operate
-//  over the None struct.
+//
+//	over the None struct.
```
Comment (iulianpascalau) [inline node/node.go:58]: Node struct :)

File: node/node.go [hunk 4]
```diff
@@ -705,7 +706,15 @@ func (n *Node) ValidateTransaction(tx *transaction.Transaction) error {
 		return err
 	}
 
-	return txValidator.CheckTxValidity(intTx)
+	err = txValidator.CheckTxValidity(intTx)
+	if errors.Is(err, process.ErrAccountNotFound) {
```
Comment (iulianpascalau) [inline node/node.go:710]: :+1:

**Review by iulianpascalau — DISMISSED** [review]

**Review by miiu96 — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.5.13-dev-config-8c685c8e00 -> adjust-account-not-found-e-e890eaa707

--- Specific errors ---

block hash does not match 9265
wrong nonce in block 3496
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  0
Nr. of all WARNS:  225
Nr. of new ERRORS:  0
Nr. of new WARNS:  1
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**